internal/npm_install/npm_install.bzl

```diff
@@ -130,6 +130,39 @@ data attribute.
         default = True,
     ),
 })
+YARN_ENVIRONMENT = [
+    "CC",
+    "CFLAGS",
+    "CXX",
+    "CXXFLAGS",
+    "JAVA_HOME",
+    "LDFLAGS",
+    "PATH",
+]
```

Comment:
I had a go at this, it seems to introduce too many merge conflicts to the point that I don't think it's too worthwhile

Reply:
This will be necessary once we apply the patches in the repo anyway: we don't want to specify a single patch that contains all the changes we made to upstream, we want a separate patch for each logical change, so we know when we can stop patching them in because upstream have merged our PRs, or because we don't need them anymore.

Reply:
good point, i didnt think about it that way, i'll split this out